Integrations/integration-Cisco-umbrella.yml

```diff
@@ -31,495 +31,1834 @@ configuration:
-
-        return hostname;
-    }
+    ''' HELPER FUNCTIONS '''
+
+    def extract_domain_name(url):
+        # Remove protocol, removes directory after the host name, returns only the host name.
+        return url.split("//")[-1].split("/")[0]
+
+
+    def http_request( api_endpoint, params_dict=None, method='GET',data_list=None):
+        req_params = {}
+        # request does not accept lists, only string/dict
+        if data_list and isinstance(data_list, list):
+            data_list = json.dumps(data_list)
+        if params_dict:
+            req_params.update(params_dict)
+        url = BASE_URL + api_endpoint
+        LOG('running %s request with url=%s\theaders=%s\nparams=%s\tdata=%s' % (method, url, DEFAULT_HEADERS, json.dumps(req_params), data_list))
+        try:
+            res = requests.request(
+                method,
+                url,
+                verify=USE_SSL,
+                params=req_params,
+                headers=DEFAULT_HEADERS,
+                data=data_list
+            )
+
+            res.raise_for_status()
+            return res.json()
+
+        except Exception, e:
+            LOG(e)
+            raise
+
+
+    def string_to_table_header(string):
+        # Checks if string, change underscore to space, capitalize every word. example: "one_two" to "One Two" (title() could work too)
+        if type(string) in STRING_TYPES:
+            return " ".join(word.capitalize() for word in string.replace("_"," ").split())
+        else:
+            return_error('The key is not a string: {}'.format(string))
+
+
+    def string_to_context_key(string):
+        # Checks if string, removes underscores, capitalize every word. example: "one_two" to "OneTwo" (title() could work too)
+        if type(string) in STRING_TYPES:
+            return "".join(word.capitalize() for word in string.split('_'))
+        else:
+            return_error('The key is not a string: {}'.format(string))
+
+
+    def date_to_ts(date):
+        # this helper function tries to parse a date time string according to a specific format
+        # if it fails, it will just output the original value
+        try:
+            ts = datetime.strptime(date[0:16],'%Y-%m-%dT%H:%M')
+        except ValueError:
+            pass
+        else:
+            if date[19]=='+':
+                ts+=timedelta(hours=int(date[20:22]),minutes=int(date[22:24]))
+            elif date[19]=='-':
+                ts-=timedelta(hours=int(date[20:22]),minutes=int(date[22:24]))
+            ts = time.mktime(ts.timetuple())
+            return str(int(ts) * 1000)
+        return date
+
+
+    def ts_to_date(ts):
+        if ts:
+        # Gets a timestamp (either str or int, either in seconds or milliseconds) and converts it to a date.
+            ts = str(ts)
+            if len(ts) > 10:
+                ts = ts[:10]
+            ts = int(ts)
+            return datetime.utcfromtimestamp(ts).strftime('%Y-%m-%dT%H:%M:%S')
+        return ts
```

Comment:
Why are you returning ts?

Reply:
In case ts is not parseable I still use it in my call, sometimes users might want to pass date time string like "2 days ago", its good for me.